### PR TITLE
Custom age distributions

### PR DESCRIPTION
**Pull request by gj262**
## Related issues and PRs

Contributes to #332 

Still to do:

- [ ] styling - the numbers column is not large enough
- [ ] update card help text to reflect the feature
- [x] waiting on feedback re: comments in #332 
- [ ] number validation (positive integers)
- [ ] age distribution value -> 'Custom'

## Description

Allow customization of age distribution.

## Impacted Areas in the application

Age distribution values.

## Testing

- age distribution values set on initial load
- change when scenario or country drop downs change
- do not change when other form fields change
- edits are passed to run()
- edits persist in the URL

### CODE REVIEW

File: src/components/Main/Scenario/SeverityTable.tsx [hunk 1]
```diff
@@ -141,6 +140,16 @@ export interface SeverityTableRow {
...
+export type SeverityTableNumberFields =
+  | 'id'
+  | 'population'
+  | 'confirmed'
+  | 'severe'
+  | 'critical'
+  | 'fatal'
+  | 'totalFatal'
+  | 'isolated'
+
```
Comment (gj262) [inline src/components/Main/Scenario/SeverityTable.tsx:152]: There is probably a better Typescript ™️  way to do this but that is beyond me.

File: src/components/Main/state/URLSerializer.ts [hunk 2]
```diff
@@ -63,3 +65,40 @@ export function deserializeScenarioFromURL(initState: State): State {
...
+
+      // prettier-ignore
+      severity.forEach((row: SeverityTableRow) => {
+        ['id', 'population', 'confirmed', 'severe', 'critical', 'fatal', 'totalFatal', 'isolated'].forEach(
+          (fieldName) => {
+            if (fieldName in row) {
+              // eslint-disable-next-line no-param-reassign
+              row[fieldName as SeverityTableNumberFields] = parseInt(
+                `${row[fieldName as SeverityTableNumberFields]}`,
+                10,
```
Comment (gj262) [inline src/components/Main/state/URLSerializer.ts:89]: There is probably a better Typescript ™️  way to do this but that is beyond me.

File: src/components/Main/Main.tsx [hunk 3]
```diff
@@ -88,19 +101,55 @@ const isRegion = (region: string): region is keyof typeof countryCaseCountData =
   return Object.prototype.hasOwnProperty.call(countryCaseCountData, region)
 }
 
+enum Provenance {
+  DEFAULT = 'DEFAULT',
```
Comment (gj262) [inline src/components/Main/Main.tsx:105]: Maybe should be called StateLifeCycle or something.

**Comment by rneher** [comment]
thanks so much. very useful.

**Comment by gj262** [comment]
Ok. So. Having the value of the age distribution field update to "Custom" is proving problematic. And I'm not available tomorrow April/3 to work on it further. So if someone else can pick this up that would be great. Otherwise I'll be back on it on April/4.

The issues are:

- the URL gets too big with the severity table packed in there
- trying to manage state among URL/reducer/useState and across two state blobs (scenarioState, severity) is hard

**Comment by gj262** [comment]
I'm going to close this in favor of #376 which avoids the issues of mixed useState/Reducer problems in this branch by putting age distribution into the scenarioState reducer so the interdependency between country selection and customized distribution can be dealt with in one sane place.

**Pull request closed without merging**